### PR TITLE
Implement basic title matching

### PR DESCRIPTION
**Pull request by palfrey**
This solves most of the core problem of #63, at least in terms of "only have episodes that match title X"

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -36,6 +37,8 @@ type Feed struct {
 	Format model.Format `toml:"format"`
 	// Custom image to use
 	CoverArt string `toml:"cover_art"`
+	// Only download episodes that match this regexp (defaults to matching anything)
+	OnlyMatching Pattern `toml:"only_matching"`
```
Comment (mxpv) [inline pkg/config/config.go]: 1. Can we drop `Pattern` and use string instead?
```go
if config.Filters.Title != "" {
    if !regexp.MatchString(config.Filters.Title, episode.Title) { ... }
}
```

2. Let's add a separate structure for the filters:
```toml:
[feeds]
  [feeds.XYZ]
  ...
  quality = "low"
  filters = { title = "regex for title here" }
```
```go
type Filters struct {
	Title string `toml:"title"`
        // More filters to be added here.
}
```

**Comment by amcgregor** [comment]
A massive question arises almost immediately.

```go
# only_matching = "pattern" # Optional Golang regexp format. If set, then only download episodes with matching titles.
```

Why on Earth is this being done within Go?  Is the goal here really to port/rewrite YouTube-DL itself?  _Really?_

https://github.com/mxpv/podsync/blob/edbc4f7f382cee6a04274782033740da53d17269/cmd/podsync/updater.go#L114

Reference YouTube-DL's in-built configuration options: (which can be combined)

* `matchtitle` — whitelisting — https://github.com/ytdl-org/youtube-dl/blob/master/youtube_dl/YoutubeDL.py#L174
* `rejecttitle` — blacklisting — https://github.com/ytdl-org/youtube-dl/blob/master/youtube_dl/YoutubeDL.py#L175

Additionally, all of the other "selection criteria" options not being passed along:

* `playliststart`, `playlistend` — range-based subsection of playlist items.
* `playlist_items` — explicit choice of desired playlist items.
* `daterange` — minimum and maximum permitted age to consider for download.
* `min_views` / `max_views` — optionally eliminate over-popular, or under-popular videos from selection.
* `match_filter` — explicit complete control via callback.

**Comment by mxpv** [comment]
> Why on Earth is this being done within Go?

Because we want to filter out XML feed items as well, in addition to skipping downloads.

**Comment by amcgregor** [comment]
> > Why on Earth is this being done within Go?
> 
> Because we want to filter out XML feed items as well, in addition to skipping downloads.

So… that reads to me like you're separately indexing episodes, even if they're not relevant, will never be downloaded, etc.  Instead of only storing relevant information, and using that information to determine inclusion in the resulting feed.  Interesting.

Which is what [these three lines](https://gist.github.com/amcgregor/666fa2b3ce673ab15ebcf2420353374b#file-render-py-L92-L94) do:

```python
infos = set(path.glob('*.info.json'))
videos = {video.with_suffix('.info.json') for video in path.glob('*.mp4')}
matched = infos & videos  # We only care about matched pairs: set intersection.

# Now, matched is the set of all .info.json files with paired, matching video files.
# Now we can generate the XML feed.
```

No video file or JSON metadata?  No inclusion in the feed.  Interesting to see differences directly attributable to bottom-up vs. top-down approaches.

File: pkg/config/config.go (hunk 1 above)
Comment (palfrey) [reply pkg/config/config.go]: Done

**Pull request merged**